build/makelib/golang.mk

```diff
@@ -171,7 +171,7 @@ go.mod.update:
 .PHONY: go.mod.check
 go.mod.check:
 	@echo === ensuring modules are tidied
-	@$(GOHOST) mod tidy
+	@$(GOHOST) mod tidy -go=1.16 && go mod tidy -go=1.17
```

Comment:
The 1.18 ci doesn't seem to be happy. You might try this flag: `-compat=1.17` instead of tidying twice.

Reply:
removed 1.18 from the list, for now, 1.18 require some other changes I'll do it in follow-up pr as discussed offline

Reply:
We don't need 1.16 here anymore, right?
```suggestion
	@$(GOHOST) mod tidy -go=1.17
```

Reply:
yes, it will be `mod tidy -compat=1.17`